cloud/grpc.go

```diff
@@ -16,8 +17,24 @@ func (c *client) withAuth(ctx context.Context) context.Context {
 	return metadata.AppendToOutgoingContext(ctx, "authorization", fmt.Sprintf("Bearer %s", c.authToken))
 }
 
+func withReqID(ctx context.Context) context.Context {
+	return metadata.AppendToOutgoingContext(ctx, "request-id", uuid.NewString())
```

Comment:
rather than always generate a new one, it would be helpful if we could override this via an environment variable (e.g. `EARTHLY_REQUEST_ID`) -- this could be useful for cases where one programatically runs earthly.

Reply:
Good thinking

Reply:
Yeah I like that idea

Reply:
I've added this, but I've kept it as a hidden flag. I think it might be useful for us debugging at some point, but I realized a lot of our commands make multiple distinct requests, and this would cause them to all have the same request ID. I'm not sure we can guarantee that wouldn't do weird things to the server in the future.